vm/vm.go

```diff
@@ -57,26 +58,34 @@ const (
 	MinAcceptedBlockWindow = 1024
 )
 
-type VM struct {
-	factory ControllerFactory
-	c       Controller
+type VM[T Controller] struct {
```

Comment:
Don't think we can do this until the controller is removed - we need the concrete type since the VM knows about the controller. We also have controller-defined apis now which the vm needs to set up so it needs to know the controller concrete type for this as well.

Reply:
Nvm - I think we have to get rid of this now that the integration test code is common. Will unstage the generics diff.